components/rangeslider/rangesliders/rangeSliderBase.js

```diff
@@ -0,0 +1,462 @@
+    set customHandleRight(value) {
+        value !== null ? this.rangeslider.setAttribute('custom-handle-right', value) : this.rangeslider.removeAttribute('custom-handle-right');
+    }
+
+    /**
+     * Handler when attribute of rangeslider is changed
+     * @param {string} name
+     * @param {any} oldValue
+     * @param {any} value
+     */
+    attributeChanged(name, oldValue, value) {
+        // We want to update the rangeslider each time value attribute has changed
+        // even though the oldValue and value are the same.
+        // The case is that I can change the value of the slider by moving it and then I want to change the value
+        // via attribute change and if it is the same as the old one the slider UI won't update.
+        if (oldValue === value && name !== 'value') return;
+        this.updateAttributeState(name, oldValue, value);
+    }
+
+    /**
+     * Will update the state properties linked with the checkbox attributes
+     * @param {string} name
+     * @param {string|boolean|array} oldValue
+     * @param {string|boolean|array} value
+     */
+    updateAttributeState(name, oldValue, value) {
+        switch (name) {
+            case 'values':
+                this.updateValuesState(oldValue, value);
+                break;
+            case 'min':
+            case 'max':
+                this.updateMinMaxState(name, value);
+                break;
+            case 'step':
+                this.updateStepState(value);
+                break;
+            case 'value':
+                this.updateValueState(value);
+                break;
+            case 'thumb':
+                this.updateThumbState(value !== null);
+                break;
+            case 'grid':
+                this.updateGridState(value !== null);
+                break;
+            case 'custom-handle':
+            case 'custom-handle-right':
+            case 'custom-handle-left':
+                this.updateCustomHandleState(name, value);
+                break;
+        }
+    }
+
+    /**
+     * Will update the target elements of the custom handles
+     * @param {string} name
+     * @param {string} value
+     */
+    updateCustomHandleState(name, value) {
+        this.initCustomhandles();
+    }
+    /**
+     * Will update the slider when min or max is changed
+     * @param {string} name
+     */
+    updateMinMaxState(name) { }
+
+    /**
+     * Will update UI of the rangeslider with its current value
+     * Useful when the step, min, max or thumb are changed
+     */
+    updateSliderPositionWithCurrentValue() { }
+
+    /**
+     * Will update UI of the rangeslider when its `values` attribute has changed
+     * @param {array} oldValue
+     * @param {array} newValue
+     */
+    updateValuesState(oldValue, newValue) { }
+
+    /**
+     * Will update UI of the rangeslider when its `value` attribute has changed
+     * @param {number|string} value
+     */
+    updateValueState(value) { }

```

Comment:
Shouldn't the implementation file contain implementations only? I understand that having the empty methods makes the class look more like both declaration and implementation, which gives us some clarity and readability, but practically has no real benefits.  Consider(but I don't think it necessary) removing them - it will somewhat reduce the complexity.

Reply:
The idea behind these methods is that they are members of an abstract class and are overridden by the children. As far as it is described in the jsdoc to document correctly these methods we need to add methods with JSdoc and then use the `@inheritdoc` tag for the methods that are overriding the ones from the abstract class.

Added error if these methods are not overridden.